nano/node/messages.cpp

```diff
@@ -1581,4 +1680,20 @@ std::size_t nano::node_id_handshake::size (nano::message_header const & header_a
 		result += sizeof (nano::account) + sizeof (nano::signature);
 	}
 	return result;
+}
+
+std::string nano::node_id_handshake::to_string () const
+{
+	std::stringstream stream;
+
+	stream << header.to_string () + "\n";
+	stream << "cookie=" + query->to_string ();
```

Comment:
The query is optional, so you need to check it has a value before trying to access it.